pandapipes/pf/pipeflow_setup.py

```diff
@@ -435,7 +437,82 @@ def create_lookups(net):
+        by the fluid and a simple rule to make sure that active nodes are connected to at least one\
+        traversed branch\
+    The result of this connectivity search is stored in the lookups (e.g. as \
+    net["_lookups"]["node_active_hydraulics"])
+
+    :param net: the pandapipes net for which to identify the connectivity
+    :type net: pandapipes.pandapipesNet
+    :param branch_pit: Internal array with branch entries
+    :type branch_pit: np.array
+    :param node_pit: Internal array with node entries
+    :type node_pit: np.array
+    :param hydraulic: flag for the mode (if True, do the check for the hydraulic simulation, \
+        otherwise for the heat transfer simulation with other considerations)
+    :type hydraulic: bool, default True
+    :return: No output
+    """
+    if hydraulic:
+        # connectivity check for hydraulic simulation
+        if get_net_option(net, "check_connectivity"):
+            nodes_connected, branches_connected = check_connectivity(net, branch_pit, node_pit)
+        else:
+            # if connectivity check is switched off, still consider oos elements
+            nodes_connected = node_pit[:, ACTIVE_ND].astype(np.bool_)
+            branches_connected = branch_pit[:, ACTIVE_BR].astype(np.bool_)
+    else:
+        # connectivity check for heat simulation (needs to consider branches with 0 velocity as
+        # well)
+        if get_net_option(net, "check_connectivity"):
+            # full connectivity check for hydraulic simulation
+            nodes_connected, branches_connected = check_connectivity(net, branch_pit, node_pit,
+                                                                     mode="heat_transfer")
+        else:
+            # if no full connectivity check is performed, all nodes that are not connected to the
+            # rest of the network wrt. flow can be identified by a more performant sum_by_group_call
+            # check for branches that are not traversed (for temperature calculation, this means
+            # that they are "out of service")
+            branches_connected = get_lookup(net, "branch", "active_hydraulics") \
+                                 & branches_connected_flow(branch_pit)
+            fn = branch_pit[:, FROM_NODE].astype(np.int32)
+            tn = branch_pit[:, TO_NODE].astype(np.int32)
+            fn_tn, flow = _sum_by_group(
+                get_net_option(net, "use_numba"), np.concatenate([fn, tn]),
+                np.concatenate([branches_connected, branches_connected]).astype(np.int32)
+            )
+            nodes_connected = np.copy(get_lookup(net, "node", "active_hydraulics"))
+            # set nodes oos that are not connected to any branches with flow > 0 (0.1 is arbitrary
+            # here, any value between 0 and 1 should work, excluding 0 and 1)
+            nodes_connected[fn_tn] = nodes_connected[fn_tn] & (flow > 0.1)
```

Comment:
Actually, I tried to make a separation here between identifying active elements (nodes / branches) for the calculation and the connectivity check itself. So, we need to identify the active elements (always!) and depending on the selected option, we do this by:
- just using the ACTIVE identifier given by the element tables (hydraulics / without connectivity check)
- performing a connectivity search from hydraulic slacks, only including ACTIVE elements in the graph search (hydraulics / with connectivity check)
- identifying all branches / nodes without flow (i.e. either set oos in hydraulics check or by checking the velocity and mass balance results) (heat_transfer / without connectivity check)
- performing a connectivity search from temperature slacks, without including elements set oos in the hydraulics check and branches with 0 flow.
This seemed most logical to me. And I think it would be most logical to communicate that branches with 0 flow can be seen as thermally inactive branches, wrt. the heat transfer these branches don't exist so that this approach is an analogy to the approach in the hydraulics.